python/hsfs/storage_connector.py

```diff
@@ -336,302 +354,273 @@ def iam_role(self):
-            return self._directory_id
-        else:
-            raise Exception(
-                "Directory ID is not supported for connector "
-                + self._storage_connector_type
-            )
+        return self._directory_id
 
     @property
     def application_id(self):
         """Application ID of the ADLS storage connector"""
-        if self._storage_connector_type.upper() == self.ADLS:
-            return self._application_id
-        else:
-            raise Exception(
-                "Application ID is not supported for connector "
-                + self._storage_connector_type
-            )
+        return self._application_id
 
     @property
     def account_name(self):
         """Account name of the ADLS storage connector"""
-        if self._storage_connector_type.upper() == self.ADLS:
-            return self._account_name
-        else:
-            raise Exception(
-                "Account name is not supported for connector "
-                + self._storage_connector_type
-            )
+        return self._account_name
 
     @property
     def container_name(self):
         """Container name of the ADLS storage connector"""
-        if self._storage_connector_type.upper() == self.ADLS:
-            return self._container_name
-        else:
-            raise Exception(
-                "Container name is not supported for connector "
-                + self._storage_connector_type
-            )
+        return self._container_name
 
     @property
     def service_credential(self):
         """Service credential of the ADLS storage connector"""
-        if self._storage_connector_type.upper() == self.ADLS:
-            return self._service_credential
-        else:
-            raise Exception(
-                "Service Credential is not supported for connector "
-                + self._storage_connector_type
+        return self._service_credential
+
+    @property
+    def path(self):
+        """If the connector refers to a path (e.g. ADLS) - return the path of the connector"""
+        if self.generation == 2:
+            return "abfss://{}@{}.dfs.core.windows.net".format(
+                self.container_name, self.account_name
             )
+        else:
+            return "adl://{}.azuredatalakestore.net".format(self.account_name)
+
+    def spark_options(self):
+        """Return prepared options to be passed to Spark, based on the additional
+        arguments.
+        """
+        return self._spark_options
+
+    def prepare_spark(self, path: Optional[str] = None):
+        """Prepare Spark to use this Storage Connector.
+
+        ```python
+        conn.prepare_spark()
+
+        spark.read.format("json").load("abfs://[bucket]/path")
+
+        # or
+        spark.read.format("json").load(conn.prepare_spark("abfs://[bucket]/path"))
+        ```
+
+        # Arguments
+            path: Path to prepare for reading from cloud storage. Defaults to `None`.
```

Comment:
The documentation here is wrong. See the `path` method above to for an example of path on abfss

Reply:
fixed